cli/generators/auth/templates/loginController.js

```diff
@@ -0,0 +1,29 @@
+import UserService from '../user/user.service';
+import jwt from 'jsonwebtoken';
+import secret from '../secret';
+
+class LoginController {
+
+  login(req, res, next) {
+    const user = req.body.user;
+    const searchCondition = {name: user.username, password: user.password};
+    const token = jwt.sign(user, secret);
+
+    UserService.findBy(searchCondition, (err, foundUser) => {
+      if (err) return next(err);
+
+      //If there's no user, then we get an empty array
+      if (foundUser.length === 0) {
+        res.status(404).json({});
```

Comment:
I think that non having elements, is not and error for returning 404

Reply:
Ok, what error code do you think we can return? My intention is to return a value different than 200 so we can realize there are no results.

Reply:
maybe 204(no content) would be more accurate

Reply:
I like it !

Reply:
In my opinion, User (or any other entity, keep on mind there is a template generator) not found functionality shouldn't be part of Login/SignUp.
On the other hand, how are you going to handle token expiration?

Reply:
Regarding the error numbers, the best one that fits in this case is `403` or even `401`.

Reply:
it is true @pablomigliasso 403 is more accurate since, we a are doing a login looking for the user into the database

Reply:
in the other hand this would be a simple strategy (as a demo) looking for the user into the database, **create a token with expiration** and returning the token and maybe name and surname, or any other public information that we may need if the user is found in the database,

Reply:
Well, it seems to be the approach of this PR is only Login/Signup. However, to complete the whole workflow maybe i have to take on mind "User Registration" and "Forgot Password". Going back to token expiration issue, what will be the behaviour when it is expired? Redirect to login? Show an alert to notify that? Renew the token (silent mode)?

Reply:
good question, what do you think @dtoranzoms if the client in this case may show an alert in the same page, but in general how to handle the 403 error should be responsibility of each client from the BE point of view

Reply:
I really do not like either 401 nor 403 for this case. Those codes are for unauthorized  (401) or for Forbidden (403). I think this case is not either case. The user just do not exist.
@mravinale regarding your question, yes. I think we should show an alert when the token expires and redirect to login when the user accepts it.

Reply:
As any authentication mechanism the last thing you need to do is to notify the user doesn't exist because it won't be secured at all.